kola/tests/kubeadm/templates.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021 Kinvolk GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package kubeadm
+
+var (
+	masterScript = `#!/bin/bash
+export CNI_VERSION={{ .CNIVersion }}
+export CRICTL_VERSION={{ .CRIctlVersion }}
+export RELEASE_VERSION={{ .ReleaseVersion }}
+export DOWNLOAD_DIR={{ .DownloadDir }}
+export RELEASE={{ .Release }}
+export PATH="${PATH}:${DOWNLOAD_DIR}"
+
+# this block redirects everything to /dev/null in order to print out only the final
+# worker config
+{
+    # create the required directory
+    # we also create a config for core user
+    # in order to have a regular access to kubectl command
+    mkdir --parent \
+        /opt/cni/bin \
+        /etc/systemd/system/kubelet.service.d \
+	${HOME}/.kube \
+	/home/core/.kube
+   
+    # we download and install the various requirements
+    # * cni plugins
+    # * crictl
+    # * kubelet service and kubeadm dropin
+    # * kubelet, kubeadm and kubectl binaries
+    curl -sSL "https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-linux-amd64-${CNI_VERSION}.tgz" |
```

Comment:
To make curl more robust one can add something like `--retry-delay 1 --retry 60 --retry-connrefused --retry-max-time 60 --connect-timeout 20`

Reply:
and `--fail` also makes sense to report an error on 404